pkg/assets/handlers.go

```diff
@@ -154,6 +154,22 @@ type WebConsoleVersion struct {
 	OpenShiftVersion  string
 }
 
+var extensionPropertiesTemplate = template.Must(template.New("webConsoleExtensionProperties").Parse(`
+window.OPENSHIFT_EXTENSION_PROPERTIES = {
+  {{ $values := .ExtensionPropertyValues }}
+  {{ range $i, $key := .ExtensionPropertyKeys }}
+  {{ if $i }},{{ end }}
+  {{ $value := index $values $i }}
+  "{{ $key | js }}": "{{ $value | js }}"
```

Comment:
The issue is not empty values, in fact I think you would want to allow
empty string for the value, empty string is technically even allowed as a
key in JSON.

The issue is that a key can be any arbitrary string, including containing
things like spaces, so they need to be wrapped in quotes.
On May 19, 2016 8:29 AM, "David Eads" <notifications@github.com> wrote:

> In pkg/assets/handlers.go
> <https://github.com/openshift/origin/pull/8934#discussion_r63868783>:
>
> > @@ -154,6 +154,22 @@ type WebConsoleVersion struct {
> >  	OpenShiftVersion  string
> >  }
> >
> > +var extensionPropertiesTemplate = template.Must(template.New("webConsoleExtensionProperties").Parse(`
> > +window.OPENSHIFT_EXTENSION_PROPERTIES = {
> > +  {{ $values := .ExtensionPropertyValues }}
> > +  {{ range $i, $key := .ExtensionPropertyKeys }}
> > +  {{ if $i }},{{ end }}
> > +  {{ $value := index $values $i }}
> > +  "{{ $key | js }}": "{{ $value | js }}"
>
> If you're worried about empty values, how about restricting them in
> validation for the object?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/8934/files/dd6185b4128407493039a8ab0e055ba3d217c370#r63868783>
>